030-Data-types/030-data-model.mdx

```diff
@@ -7,6 +7,33 @@ slug: concepts/data-model
 published: true
 ---
 
+Xata builds on top of Postgres and thus supports the [default data types](https://www.postgresql.org/docs/current/datatype.html) inherit in that platform. Xata additionally adds to that list with unique types such as `file`, `filearray` and `vector` to handle platform features like [file attachements](/docs/sdk/file-attachments) and [vector searching](/docs/sdk/vector-search). In addition, there are some added types to simplify validation of common `text` patterns such as `email` and `multiple`. Usage of these types is ultimately optional if you wish to use Xata only as a Postgres host. If you do decide to utilize them (which is the default way Xata operates through our UI), it is required that you interact with these types through the Xata SDKs.
+
+## Xata specific data types
+
+Below is a mapping of the additional data types Xata provides alongside Postgres types.
+
+| Xata data type          | PostgreSQL data type | Notes                              |
+| ----------------------- | -------------------- | ---------------------------------- |
+| [string](#string)       | text                 |                                    |
+| [text](#text)           | text                 | With constraints on maximum length |
+| [int](#int)             | bigint               |                                    |
+| [float](#float)         | double precision     |                                    |
+| [datetime](#datetime)   | timestamptz          |                                    |
+| [bool](#bool)           | boolean              |                                    |
+| [email](#email)         | text                 | Validated by Xata                  |
+| [multiple](#multiple)   | text[]               | Validated by Xata                  |
+| [file](#file)           | jsonb                | Validated by Xata                  |
```

Comment:
@sorintm I know the base type for `file` and `filearray` is `jsonb` but is that the type we show when you describe a table? I remember we have a custom type, à la `xata_file`. Can you confirm what is visible to the user for `file` and `filearray`? Thank you

Reply:
Gonna merge this for now. I'll edit later when @sorintm adds context.